### PR TITLE
Merge all potentially missing changes from 1.6 to 1.7

### PR DESCRIPTION
**Pull request by jocel1**
#Only non conflicting changes have been merged.
To be reviewed carefully to check what makes sense, and will be squashed at the end

| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | develop
| Description?  | Merge all potentially missing changes from 1.6 to 1.7
| Type?         | bug fix / improvement
| Category?     | CO
| BC breaks?    | no
| Deprecations? | no
| Fixed ticket? | 
| How to test?  |

#### Important guidelines

* Make sure [your local branch is up to date](https://help.github.com/articles/syncing-a-fork/) before commiting your changes!
* Your code MUST respect [our Coding Standards](http://doc.prestashop.com/display/PS16/Coding+Standards) (for code written in PHP, JavaScript, HTML/CSS/Smarty/Twig, SQL)!
* Your commit name MUST respect our [naming convention](http://doc.prestashop.com/display/PS16/How+to+write+a+commit+message)!

### CODE REVIEW

File: classes/order/OrderInvoice.php [hunk 1]
```diff
@@ -129,6 +129,8 @@ public function add($autodate = true, $null_values = false)
 
     public function getProductsDetail()
     {
+        $id_lang = Context::getContext()->language->id;
```
Comment (PrestaEdit) [inline classes/order/OrderInvoice.php]: This line don't need to be merge if it's the only part taken of this commit: https://github.com/jocel1/PrestaShop/commit/eb237ec154febb5c03cd6243876f93e3896e6339

$id_lang  will be never used there.

Comment (jocel1) [inline classes/order/OrderInvoice.php]: yes, this PR should definitely not be taken as is. I created it to try to figure out the amount of missing merges / PR merges in 1.7 coming from 1.6. It seems a few useful PR / JIRA fixes are definitely missing into the 1.7 branch, and we will need to thoroughly check what still needs to be merged from the 1.6 branch.

**Comment by Quetzacoalt91** [comment]
Hello,

Unfortunately, bringing the changes from the 1.6 branch made the installation failed, as we can see in the Travis logs. :disappointed: 

About
> and will be squashed at the end
Are you sure about this, keeping the commit history like this will allow us to to not merge the same commits again.

**Comment by jocel1** [comment]
Hi,
Actually I've recreated https://github.com/PrestaShop/PrestaShop/pull/7845 with a manual cherrypicking which seems to work better. Yet, it's still needs to be carefully reviewed. For this PR, I'll close it.

**Comment by Quetzacoalt91** [comment]
Okay! Thank you @jocel1

**Pull request closed without merging**